src/users/users.service.ts

```diff
@@ -361,4 +362,20 @@ export class UsersService {
         return paginate<Users>(queryBuilder, options);
     }
 
+    /**
+     * 
+     * @param empID ID of the logged in user
+     * @param newUser new User object to update old user
+     * @returns the new User object which was used to update the user
+     */
+    async editUserDetails(empID: number, newUser: Users){
+        if(empID !== newUser.employeeId){
+            throw new UnauthorizedException();
+        }
+        await this.usersRepository.save(newUser);
```

Comment:
We shouldn't allow the user to change their own role, companyId, employeeId, manager status, manager, start date, position title, recognition stats, created date, updated date, deleted date.